src/compiler/diagnosticMessages.json

```
@@ -3907,6 +3907,14 @@
         "category": "Error",
         "code": 2868
     },
+    "This JSX tag requires '{0}' to be in scope, but it could not be found.": {
+        "category": "Error",
+        "code": 2869
+    },
+    "This JSX tag requires the module path '{0}' to exist, but none could be found. Make sure you have the appropriate package installed.": {

```

```suggestion
    "This JSX tag requires the module path '{0}' to exist, but none could be found. Make sure you have types for the appropriate package installed.": {

```
? Since it's perfectly possible (common, even) to have `react` installed, but `@types/react` is still missing. Or is that a different error?

---

Whether there's a separate error, I'm not sure about - but good call on specializing the message.